admin/scripts/modules/uvod.php

```diff
@@ -140,7 +140,16 @@
     try {
         dbUpdate('uzivatele_hodnoty', $udaje, ['id_uzivatele' => $uPracovni->id()]);
     } catch (DbDuplicateEntryException $e) {
-        chyba('Uživatel se stejnou přezdívkou již existuje.');
+        // kontrola jestli šlo o změnu přezdívky nebo e-mailu (není obsaženo v chybě)
```

Comment:
Ono to v chybě je (hodil jsem si breakpoint přes XDebug a změnil jsem email na existující z jiného účtu).

`DbDuplicateEntryException` exception má tuhle strukturu

```
DbDuplicateEntryException Object
(
    [key:DbDuplicateEntryException:private] => email1_uzivatele
    [message:protected] =>  Duplicate entry 'jaroslav.tyc.83@gmail.com' for key 'email1_uzivatele' caused by [SQL...]
    [string:Exception:private] => 
    [code:protected] => 1062
    [file:protected] => /var/www/html/gamecon/model/funkce/fw-database.php
    [line:protected] => 432
    [trace:Exception:private] => Array
```

Takže stačí něco jako 

```php
if ($e->key === 'email1_uzivatele') {
    chyba('Uživatel se stejným e-mailem již existuje.');
} else if ($e->key === 'login_uzivatele') {
   chyba('Uživatel se stejným e-mailem již existuje.');
} else {
   chyba('Uživatel se stejným údajem již existuje.');
}

Reply:
Já si toho `key` nevšiml a přišlo mi neelegantní tahat to z message, díky!